src/utils/exceptions.h

```diff
@@ -54,6 +54,18 @@ class LoopCntException : public EvaluationException
   LoopCntException(const std::string& what_arg, const Location& loc) : EvaluationException(what_arg), loc(loc) {}
 };
 
+class VectorEchoStringException : public EvaluationException
+{
+public:
+  static VectorEchoStringException create() {
+    return VectorEchoStringException{STR("ERROR: Stack exhausted while trying to convert a vector to EchoString")};
```

Comment:
Small nitpick, but the `STR` macro is a helper that enables inline usage of stream operators `<<` to compose a string from variables etc.  Its not needed if there's no `<<` operators and just a single const string.